File: frontend/src/core/api/terminal.ts
import { backendApiInstance } from ".";

export type NaborIngredient = {
  id: number;
  name: string;
  price: number;
};

export type Modificator = {
  id: number;
  name: string;
  price: number;
  nabor_ingredient: NaborIngredient[];
};

export type Product = {
  id: number;
  name: string;
  categoryName: string;
  type: "tovar" | "techCart";
  image?: string;
  price: number;
  discount: boolean;
  nabor?: Modificator[];
};

export type Category = {
  id: number;
  category: string;
  image: string;
  products: Product[];
};

interface TerminalStartData {
    categories: Category[],
    mainDisplay: Product[],
    currentShift: any
}

export const getTerminalStartData = async (): Promise<TerminalStartData> => {
  const response = await backendApiInstance.get("/terminal/start");
  const { categories, mainDisplay, current_shift } = response.data.data;
  return {
    categories,
    mainDisplay,
    currentShift: current_shift
  }
};

File: backend/model/terminal.go
package model

type Product struct {
	ID           int            `json:"id"`
	Name         string         `json:"name"`
	CategoryName string         `json:"categoryName"`
	Category     int            `json:"category"`
	Type         string         `json:"type"`
	Image        string         `json:"image"`
	Price        float32        `json:"price" gorm:"column:price;not null"`
	Discount     bool           `json:"discount" gorm:"default:false"`
	Nabor        []*NaborOutput `json:"nabor"`
}

type Terminal struct {
	Categories   []*CategoryProduct `json:"categories"`
	MainDisplay  []*Product         `json:"mainDisplay"`
	CurrentShift *CurrentShift      `json:"current_shift"`
}

type CategoryProduct struct {
	ID       int        `json:"id"`
	Category string     `json:"category"`
	Image    string     `json:"image"`
	Products []*Product `json:"products"`
}
